Building OpenBLAS 0.3.20 w/ Spack using OneAPI 2022.1.0: DT_NEEDED missing libifcore

When I build `OpenBLAS@0.3.20 %oneapi@2022.1.0` with Spack, `libopenblas.so` doesn't include `libifcore` in the DT_NEEDED:

```
$> objdump -x $(spack location -i openblas)/lib/libopenblas.so | grep -i needed
  NEEDED               libm.so.6
  NEEDED               libpthread.so.0
  NEEDED               libsvml.so
  NEEDED               libirng.so
  NEEDED               libintlc.so.5
  NEEDED               libdl.so.2
  NEEDED               libiomp5.so
  NEEDED               libc.so.6
  NEEDED               ld-linux-x86-64.so.2
```

This creates issues downstream when trying to build some other packages that rely on OpenBLAS:
```
$> spack install plasma...
...
==> Installing plasma-21.8.29-y5otmtsxindyp3tuguijpg7wmypjyl7q
==> Fetching https://mirror.spack.io/_source-cache/archive/e0/e0bb4d9143c8540f9f46cbccac9ed0cbea12500a864e6954fce2fe94ea057a10.tar.gz
==> No patches needed for plasma
==> plasma: Executing phase: 'cmake'
==> plasma: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'
...
  >> 6547    /usr/bin/ld: /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/openblas-0.3.20-q6ovcjsiw3mkzigitn7q4vvidcrki424/lib/libopen
             blas.so: undefined reference to `for_cpystr'
  >> 6548    /usr/bin/ld: /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/openblas-0.3.20-q6ovcjsiw3mkzigitn7q4vvidcrki424/lib/libopen
             blas.so: undefined reference to `for_concat'
  >> 6549    /usr/bin/ld: /spack/opt/spack/linux-ubuntu20.04-x86_64/oneapi-2022.1.0/openblas-0.3.20-q6ovcjsiw3mkzigitn7q4vvidcrki424/lib/libopen
             blas.so: undefined reference to `for_cpstr'
  >> 6550    clang: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 6551    make[2]: *** [CMakeFiles/plasmatest.dir/build.make:3147: plasmatest] Error 1
...
```

You can see the linker can't find definition for symbols:
* `for_cpstr`
* `for_concat`
* `for_cpystr`

These are defined in the `libifcore.so.5` bundled with OneAPI:
```
$> nm -D /opt/intel/oneapi/compiler/2022.1.0/linux/compiler/lib/intel64_lin/libifcore.so.5 | grep for_cpstr
00000000000c1b60 T for_cpstr

$> nm -D /opt/intel/oneapi/compiler/2022.1.0/linux/compiler/lib/intel64_lin/libifcore.so.5 | grep for_cpystr
00000000000c1ed0 T for_cpystr

$> nm -D /opt/intel/oneapi/compiler/2022.1.0/linux/compiler/lib/intel64_lin/libifcore.so.5 | grep for_concat
00000000000c1ce0 T for_concat
```

I've not encountered this type of issue before so please forgive me if this isn't an issue with the OpenBLAS build. Perhaps it is an issue with the OneAPI compiler, I don't know. I can manually patch `libopenblas.so` using `patchelf` so that `libifcore.so.5` is listed as needed -- this resolves the undefined symbol problem with downstream packages that rely on OpenBLAS.

What do you think?

@wspear

--- a/f_check
+++ b/f_check
@@ -45,7 +45,7 @@ if [ -z "$compiler" ]; then
 	      pathf90 pathf95
 	      pgf95 pgf90 pgf77 pgfortran nvfortran
 	      flang egfortran
-              ifort nagfor"
+              ifort nagfor ifx"
 
     for list in $lists; do
         for p in $path; do
@@ -96,6 +96,10 @@ else
                     case "$compiler" in
                         *flang*)
                             vendor=FLANG
+                            openmp='-fopenmp'
+                            ;;
+                        *ifx*)
+                            vendor=INTEL
                             openmp='-fopenmp'
                             ;;
                         *pgf*|*nvf*)
@@ -171,7 +175,7 @@ else
                 bu=_
                 openmp='-fopenmp'
                 ;;
-            *ifort*)
+            *ifort*|*ifx*)
                 vendor=INTEL
                 bu=_
                 openmp='-fopenmp'

--- a/f_check.pl
+++ b/f_check.pl
@@ -34,7 +34,7 @@ if ($compiler eq "") {
 	      "pathf90", "pathf95",
 	      "pgf95", "pgf90", "pgf77", "pgfortran", "nvfortran",
 	      "flang", "egfortran",
-              "ifort", "nagfor");
+              "ifort", "nagfor", "ifx");
 
 OUTER:
     foreach $lists (@lists) {
@@ -89,6 +89,9 @@ if ($compiler eq "") {
 	    } else {
 		if ($compiler =~ /flang/) {
 		    $vendor = FLANG;
+		    $openmp = "-fopenmp";
+	    } elsif ($compiler =~ /ifx/) {
+		    $vendor = INTEL;
 		    $openmp = "-fopenmp";
 	    } elsif ($compiler =~ /pgf/ || $compiler =~ /nvf/) {
 		    $vendor = PGI;
@@ -176,7 +179,7 @@ if ($compiler eq "") {
 	    $openmp = "-fopenmp";
 	}
 
-	if ($compiler =~ /ifort/) {
+	if ($compiler =~ /ifort/ || $compiler =~ /ifx/) {
 	    $vendor = INTEL;
 	    $bu       = "_";
 	    $openmp = "-fopenmp";

